sdk/eventhubs/azure-messaging-eventhubs/src/main/java/com/azure/messaging/eventhubs/ObjectBatch.java

```diff
@@ -0,0 +1,62 @@
+package com.azure.messaging.eventhubs;
+
+import com.azure.core.amqp.exception.AmqpException;
+import com.azure.core.amqp.implementation.ErrorContextProvider;
+import com.azure.core.amqp.implementation.TracerProvider;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.experimental.serializer.ObjectSerializer;
+import reactor.core.publisher.Mono;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+import java.util.Objects;
+
+public final class ObjectBatch<T> extends Batch {
+    private final ClientLogger logger = new ClientLogger(ObjectBatch.class);
+    private final Object lock = new Object();
+    private final Class<T> batchType;
+    private final ObjectSerializer objectSerializer;
+
+    /**
+     *
+     * @param maxMessageSize
+     * @param partitionId
+     * @param partitionKey
+     * @param batchType
+     * @param contextProvider
+     * @param tracerProvider
+     * @param objectSerializer
+     * @param entityPath
+     * @param hostname
+     */
+    ObjectBatch(int maxMessageSize, String partitionId, String partitionKey, Class<T> batchType,
+                    ErrorContextProvider contextProvider, TracerProvider tracerProvider,
+                ObjectSerializer objectSerializer, String entityPath, String hostname) {
```

Comment:
Nit: Different indentation here (and also in previous files, there was needless changes to indentation)